### PR TITLE
Update icon docs to point to SKY UX icons 4.0.0-beta.2

### PR DESCRIPTION
**Pull request by Blackbaud-PaulCrowder**

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-skyux-icons-update](https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-skyux-icons-update)

**Review by Blackbaud-AlexKingman — DISMISSED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-skyux-icons-update](https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-skyux-icons-update)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-skyux-icons-update](https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-skyux-icons-update)

File: src/app/docs/icon/icon-docs.component.ts [hunk 1]
```diff
@@ -255,13 +254,42 @@ export class IconDocsComponent implements OnInit {
...
-      .subscribe((manifest) => {
-        this.skyuxIcons = manifest.glyphs;
-      });
+    const glyphs = this.svc.getManifest().glyphs;
+
+    this.skyuxIcons = [];
+
+    // We only want to document the non-variant or line variant of each icon and leave
+    // the solid icon undocumented for internal SKY UX use only.
+    for (const glyph of glyphs) {
```
Comment (Blackbaud-AlexKingman) [inline src/app/docs/icon/icon-docs.component.ts:266]: Will this work in IE11? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of#browser_compatibility

File: src/app/public/modules/icon/icon-resolver.service.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  constructor(manifestSvc: SkyThemeIconManifestService) {
+    // Map the icons by name for more efficient lookup.
+    for (const glyph of manifestSvc.getManifest().glyphs) {
+      this.glyphMap.set(glyph.name, glyph);
+    }
+  }
+
+  public resolveIcon(iconName: string, variant?: SkyIconVariant): string {
+    if (variant === SkyIconVariant.Solid) {
+      const solidIconName = `${iconName}-solid`;
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/icon/icon-resolver.service.ts:28]: Would we want to enforce the typing on the suffix like so?
```
const solidIconName = `${iconName}-${SkyIconVariant.Solid}`;
```

File: src/app/public/modules/icon/icon-resolver.service.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      }
+    }
+
+    if (
+      variant === SkyIconVariant.Line ||
+        // Default to the line variant if a non-variant doesn't exist, even if no
+        // variant was specified.
+        !this.glyphMap.has(iconName)
+    ) {
+      const lineIconName = `${iconName}-line`;
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/icon/icon-resolver.service.ts:41]: Would we want to enforce the typing on the suffix like so?
```
const solidIconName = `${iconName}-${SkyIconVariant.Line}`;
```

File: src/app/public/modules/icon/icon-resolver.service.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+} from './icon-variant';
+
+@Injectable()
+export class SkyIconResolverService {
+
+  private glyphMap = new Map<string, SkyThemeIconManifestGlyph>();
+
+  constructor(manifestSvc: SkyThemeIconManifestService) {
+    // Map the icons by name for more efficient lookup.
+    for (const glyph of manifestSvc.getManifest().glyphs) {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/icon/icon-resolver.service.ts:21]: Same question here about the for...of loop and IE11.

File: src/app/public/modules/icon/icon-resolver.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  it('should resolve an icon by its name', () => {
+    expect(svc.resolveIcon('no-variant')).toBe('no-variant');
+  });
+
+  it('should fall back to the non-variant if the line variant does not exist', () => {
+    expect(svc.resolveIcon('no-variant', SkyIconVariant.Line)).toBe('no-variant');
+  });
+
+  it('should fall back to the non-variant if the solid variant does not exist', () => {
+    expect(svc.resolveIcon('no-variant', SkyIconVariant.Line)).toBe('no-variant');
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/icon/icon-resolver.service.spec.ts:67]:
```suggestion
    expect(svc.resolveIcon('no-variant', SkyIconVariant.Solid)).toBe('no-variant');
```

**Review by Blackbaud-AlexKingman — COMMENTED** [review]
Few comments below...

File: src/app/docs/icon/icon-docs.component.ts (hunk 1 above)
Comment (Blackbaud-PaulCrowder) [reply src/app/docs/icon/icon-docs.component.ts:266]: Yes, it works for the same reason things like fat-arrow functions work; TypeScript is configured to transpile down to ES5-compatible JavaScript, which includes converting `for...of` to a standard `for` loop and fat arrow functions into standard functions.

File: src/app/public/modules/icon/icon-resolver.service.spec.ts (hunk 5 above)
Comment (Blackbaud-PaulCrowder) [reply src/app/public/modules/icon/icon-resolver.service.spec.ts:67]: Fixed.

File: src/app/public/modules/icon/icon-resolver.service.ts (hunk 3 above)
Comment (Blackbaud-PaulCrowder) [reply src/app/public/modules/icon/icon-resolver.service.ts:41]: After making that change, I realized I was able to simplify the logic further.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-skyux-icons-update](https://developer.blackbaud.com/skyux-indicators?_sp.spa=pr-skyux-icons-update)

**Review by Blackbaud-AlexKingman — APPROVED** [review]

**Pull request merged**